File: TEO/Mav.Services/Abstract/ILicenceService.cs
using Mav.Common.Models;
using Mav.Models.ViewModels;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace Mav.Services.Abstract
{
    public interface ILicenceService
    {
        Task<T> GetLicenceById<T>(int id);

        Task<T> GetLicenceByOrgId<T>(int orgId);

        Task<GenericResult<T>> AddLicence<T>(AddLicenceViewModel m, bool enable);

        Task<GenericResult<T>> Update<T>(UpdateLicenceViewModel m, bool enable);

        Task<GenericResult<T>> Delete<T>(int id);

        Task<GenericResult<T>> EnableAccount<T>(int orgId);

        Task<GenericResult<T>> DisableAccount<T>(int orgId);
    }
}

File: TEO/Mav.UI/Mav.Ui/src/app/services/i-licence.service.ts
import { LicenceViewModel } from "../models/admin/licence.view.model";
import { Observable } from "rxjs/Observable";
import { GenericResult } from "../models/generic.result.model";
import { AddLicenceViewModel } from "../models/admin/add-licence.view.model";
import { UpdateLicenceViewModel } from "../models/admin/update-licence.view.model";

export interface ILicenceService {

    getLicenceById(id: number): Observable<LicenceViewModel>;

    getLicenceByOrgId(orgId: number): Observable<LicenceViewModel>;

    createLicence(licence: AddLicenceViewModel): Observable<GenericResult<number>>;

    updateLicence(licence: UpdateLicenceViewModel): Observable<GenericResult<number>>;

    deleteLicence(id: number): Observable<GenericResult<number>>;
}